activejob/lib/active_job/queue_adapters/sidekiq_adapter.rb

```diff
@@ -37,7 +37,7 @@ class JobWrapper #:nodoc:
         include Sidekiq::Worker
 
         def perform(job_data)
-          Base.execute job_data
+          Base.execute job_data.merge('provider_job_id' => jid)
```

Comment:
Yeah we still need to set it in the `enqueue` method so we can access it outside like this: 
```
job = SomeJob.perform_later
jid = job.provider_job_id
```